protos/flyteidl/plugins/spark.proto

```diff
@@ -31,4 +33,24 @@ message SparkJob {
     // Domain name of your deployment. Use the form <account>.cloud.databricks.com.
     // This instance name can be set in either flytepropeller or flytekit.
     string databricksInstance = 9;
+    // The pod spec and metadata to be used as the base configuration when creating the driver Pod for this task.
+    // +optional
+    oneof driverPodValue {
+        core.K8sPod driverPod = 10;
```

Comment:
Instead of using pod specs, should we just use a google.protobuf.Struct directly (basically just the `pod_spec` field in our `K8sPod` message?

Looking at what the [their guide](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/user-guide.md#writing-driver-specification) is saying, it doesn't really look like a real K8s pod spec/template, which means we'd have to do massaging in the plugin before submitting it to the operator right?

It'd be nice if they just had a nice python class with all the right options, but it doesn't look like the `SparkConf` object is that.  So maybe just give users a nested hash map since that what it feels like that's all they provide.  Maybe one each for driver and executor.

Reply:
My thought in using the `core.K8sPod` here is to (1) pass labels and annotations as we do for container task [here](https://github.com/flyteorg/flytekit/blob/3cb79885049f9de6552d00bff1e9a57ba0d3ee0a/flytekit/core/container_task.py#L137-L141) and (2) reuse the [ToK8sPodSpec](https://github.com/flyteorg/flyteplugins/blob/6048689ca61f318ba30ac3e0df8729918314035e/go/tasks/pluginmachinery/flytek8s/pod_helper.go#L258) in flyteplugins to merge various pod configs before extracting out the relevant ones to [build](https://github.com/flyteorg/flyteplugins/blob/24a2e436819d68ab7e1dd9cfcb2c2578ebed3e65/go/tasks/plugins/k8s/spark/spark.go#L126-L162) `sparkOp.DriverSpec` and `sparkOp.ExecutorSpec`.

Reply:
oh ic.  is it always a subset though of pod spec?  or are there fields that users might want to set in the driver/executor specs that aren't in pod spec?

Reply:
Labels and annotations are not part of [PodSpec](https://pkg.go.dev/k8s.io/api/core/v1#PodSpec), but are [added to the SparkPodSpec](https://github.com/flyteorg/flyteplugins/blob/24a2e436819d68ab7e1dd9cfcb2c2578ebed3e65/go/tasks/plugins/k8s/spark/spark.go#L129-L130). Maybe we could pass the protobuf.Struct version of [PodTemplateSpec](https://pkg.go.dev/k8s.io/api/core/v1#PodTemplateSpec), which has both `PodSpec` and `ObjectMeta`, but I was err'ing on the side of consistency with [what we use for container task](https://github.com/flyteorg/flytekit/blob/3cb79885049f9de6552d00bff1e9a57ba0d3ee0a/flytekit/core/container_task.py#L137-L141). Happy to use an alternative here if more appropriate for Spark since we are just tunneling the info as long as we can send everything we need.

Reply:
okay yeah that makes sense, apologies for distracting... we can always add more fields in the future if users ask for more.  I was initially a bit concerned that their pod spec isn't a real pod spec, even though they call it that (like the cpu limit is called a `CoreLimit`, among other differences), but yeah we can easily add to the mapping as users request them in the block where we're already doing the construction of their podspec object.  thank you!